### PR TITLE
Store the entire player's list of groups instead of primary group

### PR DESCRIPTION
**Pull request by sgdc3**
#1265

### CODE REVIEW

**Comment by sgdc3** [comment]
@ljacqu could you review this + fix 4 failing tests? (I almost completely forgot how to use Junit properly xD)

File: src/test/java/fr/xephi/authme/data/limbo/LimboServiceHelperTest.java [hunk 1]
```diff
@@ -27,17 +29,17 @@
...
+        LimboPlayer oldLimbo = new LimboPlayer(oldLocation, true, Collections.singletonList("grp-old"), true, 0.1f, 0.8f);
 
         // when
         LimboPlayer result = limboServiceHelper.merge(newLimbo, oldLimbo);
 
         // then
         assertThat(result.getLocation(), equalTo(oldLocation));
         assertThat(result.isOperator(), equalTo(true));
-        assertThat(result.getGroup(), equalTo("grp-old"));
+        assertThat(result.getGroups(), equalTo(Collections.singletonList("grp-new")));
```
Comment (ljacqu) [inline src/test/java/fr/xephi/authme/data/limbo/LimboServiceHelperTest.java]: This should have rung an alarm bell — this test caught a change in behavior it was exactly meant to test ;)

**Comment by ljacqu** [comment]
@sgdc3 Great job! I merged after adding a fallback to the old "group" property so that we don't lose existing data.

**Comment by sgdc3** [comment]
Thank you ;)

**Pull request merged**